### ISSUE TITLE
step_upsample() in caret::train()

### ISSUE THREAD

**Issue by mlduarte** [tidymodels/recipes#319]
When incorporating `step_upsample()` in a recipe and calling that recipe from `caret::train()`, the following occurs:

* Successfully able to train GLM model with Accuracy metric
* Not able to train GLM model with ROC metric; the outcome variable should have had 2 levels only
> Error: task 1 failed - "auc only works for binary outcomes at this time
* Not able to train RF model

> NA not permitted in response

### Minimal, reproducible example:

```r
library(recipes)
library(caret)
library(forcats)
data(okc)
okc$height[c(1, 34234,22322,2,22)] <- NA

rec_okc <- recipe(okc) %>%
    update_role(diet, height, new_role = "predictor") %>%
    update_role(Class, new_role = "outcome") %>%
    step_medianimpute(height) %>%
     step_mutate(diet = fct_explicit_na(diet, "missing")) %>%
    step_rm(-has_role("predictor"), -all_outcomes()) %>%
    step_upsample(Class, ratio = 0.5)


# WORKS:
okc_glm1 <- train(rec_okc, 
                 data = okc, 
                 method = "glm",
                 family = "binomial",
                 trControl =  trainControl(method = "cv", 
                    number = 3,
                     classProbs = TRUE,
                     summaryFunction = defaultSummary, 
                     verboseIter = TRUE))

# Error: task 1 failed - "auc only works for binary outcomes at this time"
# this works if step_upsample() removed from recipe
okc_glm2 <- train(rec_okc, 
                 data = okc, 
                 method = "glm",
                 family = "binomial",
                 trControl =  trainControl(method = "cv", 
                     number = 3,
                     classProbs = TRUE,
                     summaryFunction = twoClassSummary, 
                     verboseIter = TRUE),
                 metric = "ROC")
# Error: model fit failed for Fold09: mtry=2 Error in randomForest.default(x, y, mtry = param$mtry, ...) : 
#  NA not permitted in response 
okc_rf <- train(rec_okc, 
                 data = okc, 
                 method = "rf",
                 ntree = 500,
                 tuneLength = 5,
                 trControl =  trainControl(method = "cv", 
                    number = 5,
                     classProbs = TRUE,
                     summaryFunction = defaultSummary, 
                     verboseIter = TRUE)
)

# Works
p_rec <- prep(rec_okc, okc, retain = TRUE)
dtrain <- juice(p_rec)
str(dtrain)
okc_rf <- train(Class ~ ., 
                 data = dtrain, 
                 method = "rf",
                 ntree = 100,
                 tuneLength = 2,
                 trControl =  trainControl(method = "cv", number = 2,
                     classProbs = TRUE,
                     summaryFunction = defaultSummary, 
                     verboseIter = TRUE)
)
```

### Session Info:

```r
sessionInfo()
```
```

R version 3.5.3 (2019-03-11)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows Server >= 2012 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] randomForest_4.6-14 forcats_0.3.0       caret_6.0-81       
[4] ggplot2_3.1.0       lattice_0.20-38     recipes_0.1.5.9000 
[7] dplyr_0.8.0.1      

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5   xfun_0.4           purrr_0.3.2       
 [4] reshape2_1.4.3     splines_3.5.3      colorspace_1.4-0  
 [7] generics_0.0.2     stats4_3.5.3       yaml_2.2.0        
[10] survival_2.43-3    prodlim_2018.04.18 rlang_0.3.3       
[13] e1071_1.7-0        ModelMetrics_1.2.2 pillar_1.3.1      
[16] glue_1.3.1         withr_2.1.2        foreach_1.4.4     
[19] plyr_1.8.4         lava_1.6.5         stringr_1.4.0     
[22] timeDate_3043.102  munsell_0.5.0      gtable_0.2.0      
[25] codetools_0.2-16   knitr_1.21         class_7.3-15      
[28] Rcpp_1.0.1         scales_1.0.0       ipred_0.9-8       
[31] stringi_1.4.3      grid_3.5.3         tools_3.5.3       
[34] magrittr_1.5       lazyeval_0.2.1     tibble_2.1.1      
[37] crayon_1.3.4       tidyr_0.8.3        pkgconfig_2.0.2   
[40] MASS_7.3-51.1      Matrix_1.2-15      data.table_1.12.0 
[43] lubridate_1.7.4    gower_0.2.0        assertthat_0.2.1  
[46] rstudioapi_0.9.0   iterators_1.0.10   R6_2.4.0          
[49] rpart_4.1-13       nnet_7.3-12        nlme_3.1-137      
[52] compiler_3.5.3    
```

**Comment by topepo** [comment]
It's a bug. The resampling indices are being computed in the data _after_ the recipe has been used on the training set. Instead of `n = 59,855` it thinks that it has `n ~ 78,000`. This induces missing data in the outcome (which is generating the error). 

I'll patch it but will have to do a lot of testing to make sure that the fix is as simple as I think it is.

**Comment by topepo** [comment]
should be working now in devel `caret`. Closed by tidymodels/recipes#1030
